source/utils/types.ts

```diff
@@ -9,10 +9,35 @@ export type NextFunction = (error?: Error | string) => void;
 
 export type IterateFunction = (options: Options) => void;
 
+export interface Response extends IncomingMessage {
+	body: string | Buffer;
+	statusCode: number;
+}
+
+export interface Timings {
+	start: number;
+	socket: number | null;
+	lookup: number | null;
+	connect: number | null;
+	upload: number | null;
+	response: number | null;
+	end: number | null;
+	error: number | null;
+	phases: {
+		wait: number | null;
+		dns: number | null;
+		tcp: number | null;
+		request: number | null;
+		firstByte: number | null;
+		download: number | null;
+		total: number | null;
+	};
+}
+
 export interface Instance {
-	methods?: Method[];
-	options?: Options;
-	handler?: (options: Options, callback: NextFunction) => void;
+	methods: Method[];
```

Comment:
This has been deprecated a while ago, no longer needed.

Reply:
In [`mergeInstances`](https://github.com/sindresorhus/got/blob/master/source/merge.ts#L60,L65) it sets the `methods` to what was passed to `mergeInstances(..., methods)`.

Reply:
It should be removed. Lemme do it.